File: problems/t/trik/trik.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int ball = 1;
    string moves;

    getline(cin, moves);

    for (size_t i = 0; i < moves.length(); i++) {
        if (moves[i] == 'A') {
            if (ball == 1) {
                ball = 2;
            }
            else if (ball == 2) {
                ball = 1;
            }
        }
        else if (moves[i] == 'B') {
            if (ball == 2) {
                ball = 3;
            }
            else if (ball == 3) {
                ball = 2;
            }
        }
        else {
            if (ball == 3) {
                ball = 1;
            }
            else if (ball == 1) {
                ball = 3;
            }
        }

    }

    cout << ball << endl;
}

File: problems/t/trik/trik.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    ball = 1
    for move in input():
        if move == 'A':
            if ball == 1:
                ball = 2
            elif ball == 2:
                ball = 1
        elif move == 'B':
            if ball == 2:
                ball = 3
            elif ball == 3:
                ball = 2
        else:
            if ball == 3:
                ball = 1
            elif ball == 1:
                ball = 3
    print(ball)

if __name__ == '__main__':
    main()